controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -145,6 +147,67 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 
 	if r.storageConsumer.GetDeletionTimestamp().IsZero() {
 
+		consumerConfigMap := &corev1.ConfigMap{}
+		consumerConfigMap.Name = r.storageConsumer.Name
+		if r.storageConsumer.Spec.ResourceNameMappingConfigMap.Name != "" {
+			consumerConfigMap.Name = r.storageConsumer.Spec.ResourceNameMappingConfigMap.Name
+		}
+		consumerConfigMap.Namespace = r.namespace
+
+		if err := r.get(consumerConfigMap); client.IgnoreNotFound(err) != nil {
+			return reconcile.Result{}, err
+		} else if consumerConfigMap.UID == "" {
+			cephCluster, err := util.GetCephClusterInNamespace(r.ctx, r.Client, r.namespace)
```

Comment:
Why is this preferable to just doing client.List with a 1 limit?

Reply:
That's what we are doing in GetCephClusterInNamespace, client.List with 1 limit. We are using it at multiple places